crates/miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -1442,6 +1442,49 @@ export.account_was_procedure_called
     # => [was_called, pad(15)]
 end
 
+#! Returns the number of procedures in the current account.
+#!
+#! Inputs:  [pad(16)]
+#! Outputs: [num_procedures, pad(15)]
+#!
+#! Where:
+#! - num_procedures is the number of procedures in the current account.
+#!
+#! Invocation: dynexec
+export.account_get_num_procedures
+    # get the number of procedures
+    exec.memory::get_num_account_procedures
+    # => [num_procedures, pad(16)]
+
+    # truncate the stack
+    swap drop
+    # => [num_procedures, pad(15)]
+end
+
+#! Returns the procedure information for the procedure at the specified index.
```

Comment:
Similar to one of the other comments: should "procedure information" be "procedure root"?